multiview/embed/mvmds.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from multiview.embed.base import BaseEmbed
+from multiview.utils.utils import check_Xs
+import warnings
+import numpy as np
+from sklearn.metrics import euclidean_distances
+
+
+class MVMDS(BaseEmbed):
+
+    """"
+    Classical Multiview Multidimensional Scaling for jointly reducing
+    the dimensions of multiple views of data. A euclidean distance matrix
+    is created for each view, double centered, and the k largest common
+    eigenvectors are returned based on the algorithm proposed by the
+    following paper:
+
+    https://www.sciencedirect.com/science/article/pii/S016794731000112X
+
+    """
+
+    def __init__(self, n_components=None):
+
+        super().__init__()
+        self.components = None
+        self.n_components = n_components
+
+    def _cpc(self, x):
+
+        """
+        Finds Stepwise Estimation of Common Principal Components as described
+        by common Trendafilov implementations based on the following paper:
+
+        https://www.sciencedirect.com/science/article/pii/S016794731000112X
+
+        Parameters
+        ----------
+
+        x: List of matrices, each with number of rows, n
+
+        it: Number of common principal component stepwise iterations
+
+        Returns
+        -------
+
+        Components: Desired Common Principal Components
+
+        """
+        n = p = x.shape[1]
+
+        views = len(x)
+
+        n_num = np.array([n] * views)/np.sum(np.array([n] * views))
+
+        Components = np.zeros((p, self.n_components))
+
+        pi = np.eye(p)
+
+        s = np.zeros((p, p))
+
+        it = 15
```

Comment:
No, don't hardcode 'it'. Leave it as a class parameter (see gcca or omnibus as an example on setting class parameters). Note that you will need code to check that it is valid, positive int.

Reply:
You can leave 15 as a default value for 'it'